serverless-plugin/dashboard.js

```diff
@@ -544,6 +557,85 @@ module.exports = function dashboard (serverless, dashboardConfig, functionDashbo
     return ruleWidgets
   }
 
+  /**
+   * Create a set of CloudWatch Dashboard widgets for Application Load Balancer services.
+   *
+   * @param {object} loadBalancerResources Object of Application Load Balancer Service resources by resource name
+   */
+  function createLoadBalancerWidgets (loadBalancerResources) {
+    const loadBalancerWidgets = []
+    for (const [logicalId] of Object.entries(loadBalancerResources)) {
+      const loadBalancerName = `\${${logicalId}.LoadBalancerName}`
+
+      const loadBalancerFullName = resolveLoadBalancerFullNameForSub(logicalId)
+      const widgetMetrics = []
+      for (const [metric, metricConfig] of Object.entries(getConfiguredMetrics(albDashConfig))) {
+        if (metricConfig.enabled) {
+          for (const stat of metricConfig.Statistic) {
+            widgetMetrics.push({
+              namespace: 'AWS/ApplicationELB',
+              metric,
+              dimensions: {
+                LoadBalancer: loadBalancerFullName
+              },
+              stat
+            })
+          }
+        }
+      }
+      if (widgetMetrics.length > 0) {
+        const metricStatWidget = createMetricWidget(
+          `ALB ${loadBalancerName}`,
+          widgetMetrics,
+          albDashConfig
+        )
+        loadBalancerWidgets.push(metricStatWidget)
+      }
+    }
+    return loadBalancerWidgets
+  }
+
+  /**
+   * Create a set of CloudWatch Dashboard widgets for Application Load Balancer Target Group services .
+   *
+   * @param {object} targetGroupResources Object of Application Load Balancer Service Target Group resources by resource name
+   * @param {object} loadBalancerResources Object of Application Load Balancer Service resources by resource name
+   */
+  function createTargetGroupWidgets (targetGroupResources, loadBalancerResources) {
+    const targetGroupWidgets = []
+    for (const [logicalId] of Object.entries(targetGroupResources, loadBalancerResources)) {
+      const loadBalancerResourceName = resolveLoadBalancerLogicalIdName(loadBalancerResources)
```

Comment:
The same issue exists here as with alarms. We need to associated the correct Target Group with the right Load Balancer. In the design, we looked at how Target Groups are associated with Load Balancers:
https://github.com/fourTheorem/slic-watch/issues/89#issue-1318309200